test/e2e/e2e_test.go

```diff
@@ -782,6 +782,204 @@ spec:
+		waitInterval              = 3 * time.Second
+	)
+
+	type node struct {
+		name   string
+		nodeIP string
+	}
+
+	var (
+		egressNode, podNode, allowedTargetNode, deniedTargetNode node
+	)
+
+	f := framework.NewDefaultFramework(testName)
+
+	// Determine what mode the CI is running in and get relevant endpoint information for the tests
+	ginkgo.BeforeEach(func() {
+		nodes, err := e2enode.GetBoundedReadySchedulableNodes(f.ClientSet, 2)
+		framework.ExpectNoError(err)
+		if len(nodes.Items) < 2 {
+			framework.Failf("Test requires >= 2 Ready nodes, but there are only %v nodes", len(nodes.Items))
+		}
+		ips := e2enode.CollectAddresses(nodes, v1.NodeInternalIP)
+		egressNode = node{
+			name:   nodes.Items[1].Name,
+			nodeIP: ips[1],
+		}
+		podNode = node{
+			name:   nodes.Items[0].Name,
+			nodeIP: ips[0],
+		}
+		allowedTargetNode = node{
+			name: allowedNodeName,
+		}
+		allowedTargetNode.nodeIP = createClusterExternalContainer(allowedTargetNode.name, "docker.io/httpd", []string{"--network", ciNetworkName, "-P"})
+		deniedTargetNode = node{
+			name: deniedNodeName,
+		}
+		deniedTargetNode.nodeIP = createClusterExternalContainer(deniedTargetNode.name, "docker.io/httpd", []string{"--network", ciNetworkName, "-P"})
+	})
+
+	ginkgo.AfterEach(func() {
+		deleteClusterExternalContainer(allowedTargetNode.name)
+		deleteClusterExternalContainer(deniedTargetNode.name)
+	})
+
+	ginkgo.It("Should validate the egress IP functionality against remote hosts with egress firewall applied", func() {
+		podHTTPPort := "8080"
+		podName := "e2e-egressip-pod"
+		podEgressLabel := map[string]string{
+			"wants": "egress",
+		}
+		command := []string{"/agnhost", "netexec", fmt.Sprintf("--http-port=%s", podHTTPPort)}
+		frameworkNsFlag := fmt.Sprintf("--namespace=%s", f.Namespace.Name)
+
+		ginkgo.By("0. Add the \"k8s.ovn.org/egress-assignable\" label to one nodes")
+		framework.AddOrUpdateLabelOnNode(f.ClientSet, egressNode.name, "k8s.ovn.org/egress-assignable", "dummy")
+
+		podNamespace := f.Namespace
+		podNamespace.Labels = map[string]string{
+			"name": f.Namespace.Name,
+		}
+		updateNamespace(f, podNamespace)
+
+		ginkgo.By("1. Create an EgressIP object with one egress IP defined")
+		dupIP := func(ip net.IP) net.IP {
+			dup := make(net.IP, len(ip))
+			copy(dup, ip)
+			return dup
+		}
+		// Assign the egress IP without conflicting with any node IP,
+		// the kind subnet is /16 or /64 so the following should be fine.
+		egressNodeIP := net.ParseIP(egressNode.nodeIP)
+		egressIP := dupIP(egressNodeIP)
+		egressIP[len(egressIP)-2]++
+
+		var egressIPConfig = `apiVersion: k8s.ovn.org/v1
+kind: EgressIP
+metadata:
+    name: egressip
+spec:
+    egressIPs:
+    - ` + egressIP.String() + `
+    podSelector:
+        matchLabels:
+            wants: egress
+    namespaceSelector:
+        matchLabels:
+            name: ` + f.Namespace.Name + `
+`
+
+		if err := ioutil.WriteFile(egressIPYaml, []byte(egressIPConfig), 0644); err != nil {
```

Comment:
can we do this without writing to disk?

Reply:
As I see it: not without some weird go-bash-combined hackery. `kubectl` does allow you to create an object from stanza which is written to the process STDIN file descriptor. But doing that from here is severe overkill in my opinion. The files are cleaned once the tests are done anyways.